### PR TITLE
[Infochannel] Rewrite: New counters, customize names, better rate limits

### PR DESCRIPTION
**Pull request by ASSASSIN0831**

### CODE REVIEW

File: infochannel/infochannel.py [hunk 1]
```diff
@@ -261,16 +261,6 @@ async def _infochannelset_rolecount(self, ctx: commands.Context, role: discord.R
...
-    #delete later
-    @infochannelset.command(name="cleardata")
-    async def _infochannelset_cleardata(self, ctx: commands.Context):
-        """
-        Clears the the servers data in case of corruption
-        """
-        guild = ctx.guild
-        await self.config.guild(guild).clear()
-        await ctx.send("The data for this server is cleared.")
-
```
Comment (ASSASSIN0831) [inline infochannel/infochannel.py]: I added that to help during debugging and forgot to delete it at the end. Whoops

**Comment by ASSASSIN0831** [comment]
I was wondering if there should be some sort of hardcoded limit to the amount of role counters you can add. I feel like with too many, you could run into a rate limit issue

**Review by bobloy — CHANGES_REQUESTED** [review]
Firstly, thank you so much for this work. I greatly appreciate it, and I'm sure the users of this cog will too.

I'm starting the review on this, but there are a few issue right off the bat.

The first and easiest to fix is this should follow the `black` formatting guidelines with line length of 99.
I recommend `pip install black` then `black <file_path> -l 99` to apply these formatting changes.

Secondly, your fear about rate limits are correct. Due to the number of events this is listening to, it's likely better off ignoring the events entirely and running on a schedule instead.

With your permission, I'd like to take this PR and make the necessary modifications myself.

**Comment by ASSASSIN0831** [comment]
I ran the black formatting command you gave and have now committed that change. Also, with the change to get rid of the listeners, remember to not entirely get rid of the on_guild_role_delete listener because it's used to delete the role's channel if the role gets deleted.

**Comment by bobloy** [comment]
I've applied the rate-limiting features I was interested in. Ended up using listeners still.

Also refactor much of the code, so this is now a complete rewrite.

Incompatible with older versions, so channels will need to be deleted and re-setup.

**Pull request merged**